src/qos_hex/src/lib.rs

```diff
@@ -140,7 +140,7 @@ pub fn decode_to_buf(raw_s: &str, buf: &mut [u8]) -> Result<(), HexError> {
 		1 => return Err(HexError::LengthOne),
 		_ => {
 			if &raw_s.as_bytes()[0..2] == b"0x" {
-				raw_s[2..].as_bytes()
+				&raw_s.as_bytes()[2..]
```

Comment:
are these actually equivalent? I don't think so, but it seems like that's what we want based on the line above: "if the first two bytes are `b"0x"`, then skip these two bytes". I think we were skipping the first two string characters before.